### PR TITLE
Upgrade phpunit to 7.X

### PR DESCRIPTION
**Pull request by mikeymike**
### Description
Upgrade PHPUnit to 7

Currently only resolved failing unit tests and I'm fully expecting functional / integration tests to have some issues too but thought I'd start with this and check the tests in Travis

### Fixed Issues (if relevant)
https://github.com/magento-engcom/zf1-php-7.2-support/issues/6

### Manual testing scenarios
`phpunit -c dev/tests/unit/phpunit.xml.dist`

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: lib/internal/Magento/Framework/TestFramework/Unit/Matcher/MethodInvokedAtIndex.php [hunk 1]
```diff
@@ -6,10 +6,13 @@
...
 namespace Magento\Framework\TestFramework\Unit\Matcher;
 
+use PHPUnit\Framework\ExpectationFailedException;
+use PHPUnit\Framework\MockObject\Invocation;
+
 /**
  * Class MethodInvokedAtIndex
  * Matches invocations per 'method' at 'position'
- * Example:
+ * Example:§
```
Comment (buskamuza) [inline lib/internal/Magento/Framework/TestFramework/Unit/Matcher/MethodInvokedAtIndex.php]: Is paragraph symbol expected here?

**Review by buskamuza — CHANGES_REQUESTED** [review]

**Comment by buskamuza** [comment]
Thanks for submitting the PR!
I see that some tests failed on Travis and there is one small question in the PR.
Also, I changed the target branch to `2.3-develop` as all the work from `libs-upgrade` has just been merged to `2.3-develop` recently and we'll stop the work in `libs-upgrade`.

File: lib/internal/Magento/Framework/TestFramework/Unit/Matcher/MethodInvokedAtIndex.php (hunk 1 above)
Comment (mikeymike) [reply lib/internal/Magento/Framework/TestFramework/Unit/Matcher/MethodInvokedAtIndex.php]: whoops, ammended and pushed

**Comment by mikeymike** [comment]
@buskamuza thanks for the review, I'll take a look at the failing tests

**Comment by mikeymike** [comment]
@buskamuza I'm not really sure why the last run is failing, potentially some edge case of some kind... would probably pass if rerun 🤔 

Anyway I just read on slack that PHPUnit upgrade has already been completed by someone else so I guess this might have been a waste of time... I'll leave it for you to handle as needed 👍

File: lib/internal/Magento/Framework/TestFramework/Unit/Matcher/MethodInvokedAtIndex.php [hunk 2]
```diff
@@ -18,30 +21,83 @@
...
  */
-class MethodInvokedAtIndex extends \PHPUnit_Framework_MockObject_Matcher_InvokedAtIndex
+class MethodInvokedAtIndex implements \PHPUnit\Framework\MockObject\Matcher\Invocation
 {
     /**
      * @var array
      */
     protected $indexes = [];
 
+    /**PHPUnit_Framework_MockObject_Matcher_InvokedAtIndex
```
Comment (slavvka) [inline lib/internal/Magento/Framework/TestFramework/Unit/Matcher/MethodInvokedAtIndex.php:31]: Typo
